samples/gbmc_socket.py

```diff
@@ -0,0 +1,326 @@
+        packet = length + data
+        self.conn.send(packet)
+
+
+def calc_tail_correction(box):
+    '''
+    Calculates the tail correction to the energies of the box.
+    '''
+    return box.n_particles**2 * 8.0 / 3.0 * np.pi / box.box_l**3 * \
+        (1.0 / 3.0 * np.power(lj_cutoff, -9) - np.power(lj_cutoff, -3))
+
+
+def calc_shift_correction(box):
+    '''
+    Calculates the shift correction to the energies of the box.
+    '''
+    return -1.0 * box.n_particles**2 * 8.0 / 3.0 * np.pi / \
+        box.box_l**3 * np.power(lj_cutoff, 3) * lj_shift
+
+
+# init socket
+s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+s.bind((HOST, PORT))
+s.listen(number_of_clients)
+
+boxes = []
+p = []
+
+random.seed(seed)
+
+# start clients and connections
+for i in range(number_of_clients):
+    boxes.append(box())
+    p.append(subprocess.Popen(
+        "../pypresso ./gbmc_client.py -n {} -lj {} {} {} {} -s {} -bl {}".format(boxes[i].n_particles,
+                                                                                 lj_epsilon,
+                                                                                 lj_sigma,
+                                                                                 lj_cutoff,
+                                                                                 lj_shift,
+                                                                                 random.randint(
+                                                                                     0, np.iinfo(np.int32).max),
+                                                                                 boxes[i].box_l),
+        shell=True))
+
+    boxes[i].conn, boxes[i].adr = s.accept()
+    boxes[i].send_data(pickle.dumps([MSG_START, 0]))
+
+    boxes[i].recv_energy()
+    boxes[i].energy_old = boxes[i].energy + \
+        calc_tail_correction(boxes[i]) + calc_shift_correction(boxes[i])
+
+# observables
+densities = [[], []]
+list_indices = []
+accepted_steps = 0
+accepted_steps_move = 0
+accepted_steps_volume = 0
+accepted_steps_exchange = 0
+move_steps_tried = 0
+volume_steps_tried = 0
+exchange_steps_tried = 0
+
+# rand_box defines on which box the move acts on.
+rand_box = 0
+
+# Monte-Carlo loop
+for i in range(steps):
+    print("\rIntegrating: {0:2d}%".format(int(100 * i / steps)), end='', flush=True)
+    # rand defines which move to make.
+    rand = random.random()
+
+    # chose step and send the command to execute to the clients, then recieve
+    # the energies.
+    if(rand <= move_chance):
+        if(random.randint(0, global_num_particles - 1) < boxes[0].n_particles):
+            rand_box = 0
+        else:
+            rand_box = 1
+        boxes[rand_box].send_data(pickle.dumps([MSG_MOVE_PART, 0]))
+        boxes[rand_box].recv_energy()
+
+        move_steps_tried += 1
+
+    elif(rand <= move_chance + volume_chance):
+        boxes[0].box_l_old = boxes[0].box_l
```

Comment:
I think it would be better to split this whole segment into functions. It's currently very hard to understand what happens when. If you give the blocks a name, it will greatly improve readability